src/modules/compliance/src/lib/Engine.cpp

```diff
@@ -0,0 +1,324 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#include "Engine.h"
+
+#include <CommonUtils.h>
+#include "Evaluator.h"
+#include "Base64.h"
+
+#include "parson.h"
+#include <string>
+#include <vector>
+#include <map>
+#include <sstream>
+
+namespace compliance
+{
+    static constexpr const char* cLogFile = "/var/log/osconfig_compliance.log";
+    static constexpr const char* cRolledLogFile = "/var/log/osconfig_compliance.bak";
+    static constexpr const char* cModuleInfo = "{\"Name\": \"Compliance\","
+        "\"Description\": \"Provides functionality to audit and remediate Security Baseline policies on device\","
+        "\"Manufacturer\": \"Microsoft\","
+        "\"VersionMajor\": 2,"
+        "\"VersionMinor\": 0,"
+        "\"VersionInfo\": \"Dilithium\","
+        "\"Components\": [\"Compliance\"],"
+        "\"Lifetime\": 2,"
+        "\"UserAccount\": 0}";
+
+    Engine::Engine(void* log) noexcept : mLog{ log }
+    {
+    }
+
+    Engine::Engine() noexcept : mLog{ OpenLog(cLogFile, cRolledLogFile) }, mLocalLog{ true }
+    {
+    }
+
+    Engine::~Engine()
+    {
+        if(mLocalLog)
+        {
+            CloseLog(&mLog);
+        }
+    }
+
+    void Engine::setMaxPayloadSize(unsigned int value) noexcept {
```

Comment:
>{ [](http://example.com/codeflow?start=64&length=1)

Please move opening curly brace { on the next line.

In this project, in all C and C++ code, we do:

foo
{
    ...
}

and not:

foo (
  ....
}

Here and elsewhere.